python/cudf/cudf/core/series.py

```diff
@@ -960,23 +956,61 @@ def memory_usage(self, index=True, deep=False):
         return sum(super().memory_usage(index, deep).values())
 
     def __array_function__(self, func, types, args, kwargs):
-        handled_types = [cudf.Series]
-        for t in types:
-            if t not in handled_types:
+        if "out" in kwargs or any(not issubclass(t, (Series,)) for t in types):
+            return NotImplemented
+
+        fname = func.__name__
+
+        try:
+            # Apply a Series method if one exists.
+            cudf_ser_func = getattr(Series, fname, None)
+            if cudf_ser_func:
```

Comment:
This is amazing. Two walruses. Walrii?

With this suggestion and the one below it, you can remove the variable `fname` again.
```suggestion
            if cudf_ser_func := getattr(Series, func.__name__, None):
```